src/Block/Service/AbstractBlockService.php

```diff
@@ -36,19 +36,38 @@ abstract class AbstractBlockService implements BlockServiceInterface
     protected $templating;
 
     /**
-     * @param string          $name
-     * @param EngineInterface $templating
+     * @param EngineInterface|string $templatingOrDeprecatedName
+     * @param EngineInterface        $templating
      */
-    public function __construct($name = null, EngineInterface $templating = null)
+    public function __construct($templatingOrDeprecatedName = null, EngineInterface $templating = null)
     {
-        if (null === $name || null === $templating) {
-            @trigger_error(
-                'The $name and $templating parameters will be required fields with the 4.0 release.',
-                E_USER_DEPRECATED
-            );
+        if ($templatingOrDeprecatedName instanceof EngineInterface) {
+            $this->templating = $templatingOrDeprecatedName;
+        } else {
+            $this->name = $templatingOrDeprecatedName;
+            $this->templating = $templating;
         }
 
-        $this->name = $name;
+        if (0 !== strpos(static::class, __NAMESPACE__.'\\')) {
+            if (null === $templatingOrDeprecatedName) {
```

Comment:
Can't this condition merged with the next one? The goal for using `gettype()` is to handle "NULL" or "string" in the same check (`!$templatingOrDeprecatedName instanceof EngineInterface`).

Reply:
Good catch!